spec/dfe/analytics/requests_spec.rb

```diff
@@ -111,4 +111,50 @@ def index
       end).to have_been_made
     end
   end
+
+  context 'rack page caching' do
+    let!(:event) do
+      { environment: 'test',
+        event_type: 'web_request',
+        request_user_agent: nil,
+        request_method: 'GET',
+        request_path: '/unauthenticated_example',
+        request_query: [],
+        request_referer: nil,
+        anonymised_user_agent_and_ip: '12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0',
+        response_content_type: 'text/html; charset=utf-8',
+        response_status: 304 }
+    end
+
+    # Modify controller so that web request events are not handled by the controller
+    # Otherwise we'll get 2 web request events in the test
+    before do
+      unauthenticated_controller = Class.new(ApplicationController) do
+        def index
+          render plain: 'hello'
+        end
+      end
+
+      stub_const('TestUnauthenticatedController', unauthenticated_controller)
+    end
+
+    it 'serves a cached page from rack middleware' do
+      # Indicate to rack middleware that this page is cached
+      allow(DfE::Analytics).to receive(:rack_page_cached?).and_return(true)
```

Comment:
It would be cool to get `ActionDispatch::Static` to do the right thing so we could test this more realistically but I guess we could add that later.

Reply:
To do that we'd have to implement rails page caching in one of the pages in the dummy App. It's easy enough to do, but might be fiddly get it to work in rspec. I'll open a ticket in the DI team backlog to do this in the future.